### ISSUE TITLE
How to properly stop server?

### ISSUE THREAD

**Issue by VergilUa** [alvyxaz/barebones-masterserver#101]
Hi guys. I'm quite new to the MSF and I've been writing a custom console handler for MSF. I've ran into an issue with stopping server properly. It seems that calling Application.Quit directly causes WebSocket exceptions:

Console handler log:

> 5/29/2017 1:35:02 PM|Error|WebSocketServer.Stop:0|This operation is not available in: ready
> [Info | MasterServerBehaviour] Started on port: 5000
> [Info | ConnectionToMaster] Connected to: 127.0.0.1:5000
>  shutdown
> Save called.
> 5/29/2017 1:35:09 PM|Fatal|WebSocketServer.receiveRequest:0|System.NullReferenceException: Object reference not set to an instance of an object
>                              at System.Net.Sockets.TcpListener.EndAcceptTcpClient (IAsyncResult asyncResult) ?[0x00000] in <filename unknown>:0
>                              at WebSocketSharp.Server.WebSocketServer.receiveRequest () [0x00000] in <filename unknown>:0
> 5/29/2017 1:35:09 PM|Error|WebSocketServer.Stop:0|This operation is not available in: stop

Shutdown method:
`private void Shutdown() {
			Application.Quit();
}`

Server is run via build cmd line -> server.exe -msfStartMaster -batchmode -nographics

Seems like there's plenty of tutorials about starting server, but there's like 0 mentions about how to stop it in wiki.

Also, I've tried obtaining MasterServerBehavior and calling .StopServer(), but it still throws exceptions.

**Comment by alvyxaz** [comment]
Hey!

Yeah, it's been bugging me for a while, but I just started to ignore them.

I didn't have a case where it caused any problems (other than the ugly log). In my other projects (not unity related), I just disable internal websocket logging, as it never gave any valuable information.

In this framework, there are two ways to go about it:
* Disable the internal logging 
* Introduce some other mechanisms to stop the server (with waiting mechanisms that would wait until sockets are closed) - I did try that, but it ended up causing more trouble than it was worth.

If you have suggestions, I'd love to hear them ;)
